src/widgets/index/__tests__/index-test.ts

```diff
@@ -651,6 +652,103 @@ See documentation: https://www.algolia.com/doc/api-reference/widgets/index-widge
     });
   });
 
+  describe('createURL', () => {
+    it('default url returns #', () => {
+      const instance = index({ indexName: 'indexName' });
+      const searchBox = createSearchBox();
+      const pagination = createPagination();
+
+      instance.addWidgets([searchBox, pagination]);
+
+      instance.init(createInitOptions());
+
+      expect(instance.createURL(new SearchParameters())).toEqual('#');
+    });
+
+    it('calls the createURL of routing', () => {
+      const instance = index({ indexName: 'indexName' });
+      const searchBox = createSearchBox();
+      const pagination = createPagination();
+
+      instance.addWidgets([searchBox, pagination]);
+
+      instance.init(
+        createInitOptions({
+          instantSearchInstance: createInstantSearch({
+            // @ts-ignore
```

Comment:
Is it still needed as it's called at L679?

Reply:
yes, this function is meant to return a string, but an object is easier to assert on